optuna/samplers/nsgaiii.py

```diff
@@ -0,0 +1,542 @@
+from collections import defaultdict
+import hashlib
+import itertools
+import math
+from typing import Any
+from typing import Callable
+from typing import DefaultDict
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Sequence
+from typing import Tuple
+import warnings
+
+import numpy as np
+
+import optuna
+from optuna.distributions import BaseDistribution
+from optuna.exceptions import ExperimentalWarning
+from optuna.samplers._base import _CONSTRAINTS_KEY
+from optuna.samplers._base import _process_constraints_after_trial
+from optuna.samplers._base import BaseSampler
+from optuna.samplers._random import RandomSampler
+from optuna.samplers._search_space import IntersectionSearchSpace
+from optuna.samplers.nsgaii._crossover import perform_crossover
+from optuna.samplers.nsgaii._crossovers._base import BaseCrossover
+from optuna.samplers.nsgaii._crossovers._uniform import UniformCrossover
+from optuna.samplers.nsgaii._sampler import _constrained_dominates
+from optuna.study import Study
+from optuna.study._multi_objective import _dominates
+from optuna.trial import FrozenTrial
+from optuna.trial import TrialState
+
+
+# Define key names of `Trial.system_attrs`.
+_GENERATION_KEY = "nsga3:generation"
+_POPULATION_CACHE_KEY_PREFIX = "nsga3:population"
+
+
+class NSGAIIISampler(BaseSampler):
+    """Multi-objective sampler using the NSGA-III algorithm.
```

Comment:
The reference for `NSGAIIISampler` was not rendered. I think this is because it is not listed in https://github.com/optuna/optuna/blob/master/docs/source/reference/samplers/index.rst. Could you add `NSGAIIISampler` to the target of `autosummary `?